src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/participationhistory/restapi/CourseParticipationIntegrationTest.kt

```diff
@@ -200,38 +204,52 @@ constructor(
       createdAt = LocalDateTime.MAX.format(DateTimeFormatter.ISO_DATE_TIME),
     )
 
-    courseParticipationFromUpdate.shouldBeEqualToIgnoringFields(expectedCourseParticipation, CourseParticipation::createdAt)
-    LocalDateTime.parse(courseParticipationFromUpdate.createdAt) shouldBeGreaterThanOrEqualTo startTime
-    getCourseParticipation(courseParticipationId).shouldBeEqualToIgnoringFields(expectedCourseParticipation, CourseParticipation::createdAt)
+    updated.shouldBeEqualToIgnoringFields(expected, CourseParticipation::createdAt)
+    LocalDateTime.parse(updated.createdAt) shouldBeGreaterThanOrEqualTo startTime
+    getCourseParticipation(created.id).shouldBeEqualToIgnoringFields(expected, CourseParticipation::createdAt)
   }
 
   @Test
   fun `Finding course participations by prison number should return 200 with matching entries`() {
     val prisonNumber = randomPrisonNumber()
-    val ids = getCourseIds().map { createCourseParticipation(minimalCourseParticipation(it, prisonNumber)).id }.toSet()
-    ids.shouldNotBeEmpty()
 
-    val otherIds = getCourseIds().map { createCourseParticipation(minimalCourseParticipation(it, randomPrisonNumber())) }.toSet()
+    val expectedCourseIdsForPrisonNumber = getCourseIds().map {
+      createCourseParticipation(
+        CreateCourseParticipation(
+          courseId = it,
+          prisonNumber = prisonNumber,
+        )
+      ).id }.toSet()
 
-    otherIds.intersect(ids).shouldBeEmpty()
+    expectedCourseIdsForPrisonNumber.shouldNotBeEmpty()
 
-    val courseIdsForPrisonNumber = webTestClient
-      .get()
-      .uri("/people/{prisonNumber}/course-participations", prisonNumber)
-      .headers(jwtAuthHelper.authorizationHeaderConfigurer())
-      .accept(MediaType.APPLICATION_JSON)
-      .exchange()
-      .expectStatus().isOk
-      .expectBody<List<CourseParticipation>>()
-      .returnResult().responseBody!!.map { it.id }.toSet()
+    val courseIdsForRandomPrisonNumber = getCourseIds().map {
+      createCourseParticipation(
+        CreateCourseParticipation(
+          courseId = it,
+          prisonNumber = randomPrisonNumber(),
+        )
+      ) }.toSet()
+
+    courseIdsForRandomPrisonNumber.intersect(expectedCourseIdsForPrisonNumber).shouldBeEmpty()
 
-    courseIdsForPrisonNumber shouldBe ids
+    val retrievedCourseIdsForPrisonNumber = getCourseParticipationsForPrisonNumber(prisonNumber).map { it.id }.toSet()
+
+    retrievedCourseIdsForPrisonNumber shouldBe expectedCourseIdsForPrisonNumber
   }
 
   @Test
   fun `Finding course participations by random prison number should return 200 with no matching entries`() {
     val prisonNumber = randomPrisonNumber()
-    getCourseIds().map { createCourseParticipation(minimalCourseParticipation(it, prisonNumber)) }
+
+    getCourseIds().map {
```

Comment:
as discussed on slack, this can be `foreach` (even though it's not a change you've made!)